api/tests/Unit/UserAuthorizationScopeTest.php

```diff
@@ -0,0 +1,263 @@
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(null);
+
+        $userIds = User::authorizedToViewBasicInfo()->get()->pluck('id');
+        assertEqualsCanonicalizing([], $userIds->toArray());
+    }
+
+    // an applicant should be able to view just themselves
+    public function testViewBasicAsApplicant(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn($this->user1);
+
+        $userIds = User::authorizedToViewBasicInfo()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->user1->id,
+        ], $userIds->toArray());
+    }
+
+    // an pool operator should be able to view themselves and any users with submitted candidates in their pools
+    public function testViewBasicAsPoolOperator(): void
+    {
+        $poolOperator = User::factory()
+            ->asPoolOperator($this->pool1->team->name)
+            ->create();
+        Auth::shouldReceive('user')
+            ->andReturn($poolOperator);
+
+        // can also see the user with an application submitted to team 1's pool
+        $userIds = User::authorizedToViewBasicInfo()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $poolOperator->id,
+            $this->user1->id,
+        ], $userIds->toArray());
+    }
+
+    // a request responder should be able to view any user
+    public function testViewBasicAsRequestResponder(): void
+    {
+        $requestResponder = User::factory()
+            ->asRequestResponder()
+            ->create();
+        Auth::shouldReceive('user')
+            ->andReturn($requestResponder);
+
+        $userIds = User::authorizedToViewBasicInfo()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->platformAdmin->id,
+            $this->user1->id,
+            $this->user2->id,
+            $requestResponder->id,
+        ], $userIds->toArray());
+    }
+
+    // a community manager can see any basic info
+    public function testViewBasicAsCommunityManager(): void
+    {
+        $communityManager = User::factory()
+            ->asCommunityManager()
+            ->create();
+        Auth::shouldReceive('user')
+            ->andReturn($communityManager);
+
+        $userIds = User::authorizedToViewBasicInfo()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->platformAdmin->id,
+            $this->user1->id,
+            $this->user2->id,
+            $communityManager->id,
+        ], $userIds->toArray());
+    }
+
+    // a platform admin should be able to view any user
+    public function testViewBasicAsPlatformAdmin(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn($this->platformAdmin);
+
+        $userIds = User::authorizedToViewBasicInfo()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->platformAdmin->id,
+            $this->user1->id,
+            $this->user2->id,
+        ], $userIds->toArray());
+    }
```

Comment:
Could you add the reminder here as well? `TODO: add tests for new roles as part of #10609`

Reply:
[extra comments](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11138/commits/dbda192c713a308da0628b1501ca454a706f879a)